tests/test_preprocessing.py

```diff
@@ -52,6 +52,22 @@
     await program_runner.join()
 
 
+@mark.asyncio
+async def test_get_bit(test_preprocessing):
+    n, t = 4, 1
+    num_bits = 10
+    test_preprocessing.generate("bits", n, t)
+
+    async def _prog(ctx):
+        for _ in range(num_bits):
+            x_sh = test_preprocessing.elements.get_bit(ctx)
+            assert await x_sh.open() == 0 or 1
```

Comment:
Use ShareArray open instead of share open?

```
x = context.ShareArray([i.v for i in test_preprocessing.elements.get_bit(ctx)])
```

Reply:
In addition, I think that `a == 0 or 1` isn't quite doing what you want it to do--
```
>>> for a in range(4):
...     print(f"{a == 0 or 1} => {a == 0 or 1}")
... 
True => True
1 => 1
1 => 1
1 => 1
```
This is effectively: 
```
assert (await x_sh.open() == 0) or 1
```
This will always be `True`, as if `x_sh.open() == 4`, for example, this will still pass the test.

Reply:
Nice catch!